stdlib/pathlib/_abc.pyi

```diff
@@ -0,0 +1,192 @@
+    @overload
+    def join(self, path: StrPath, *paths: StrPath) -> str: ...
+    @overload
+    def join(self, path: BytesPath, *paths: BytesPath) -> bytes: ...
+    @overload
+    def split(self, path: PathLike[AnyStr]) -> tuple[AnyStr, AnyStr]: ...
+    @overload
+    def split(self, path: AnyOrLiteralStr) -> tuple[AnyOrLiteralStr, AnyOrLiteralStr]: ...
+    @overload
+    def splitdrive(self, path: PathLike[AnyStr]) -> tuple[AnyStr, AnyStr]: ...
+    @overload
+    def splitdrive(self, path: AnyOrLiteralStr) -> tuple[AnyOrLiteralStr, AnyOrLiteralStr]: ...
+    @overload
+    def normcase(self, path: PathLike[AnyStr]) -> AnyStr: ...
+    @overload
+    def normcase(self, path: AnyOrLiteralStr) -> AnyOrLiteralStr: ...
+    def isabs(self, path: StrOrBytesPath) -> bool: ...
+
+class PurePathBase:
+    parser: ClassVar[types.ModuleType | ParserBase]
+    def __init__(self, path: StrPath, *paths: StrPath) -> None: ...
+    def with_segments(self, *pathsegments: StrPath) -> Self: ...
+    def as_posix(self) -> str: ...
+    @property
+    def drive(self) -> str: ...
+    @property
+    def root(self) -> str: ...
+    @property
+    def anchor(self) -> str: ...
+    @property
+    def name(self) -> str: ...
+    @property
+    def suffix(self) -> str: ...
+    @property
+    def suffixes(self) -> list[str]: ...
+    @property
+    def stem(self) -> str: ...
+    def with_name(self, name: str) -> Self: ...
+    def with_stem(self, stem: str) -> Self: ...
+    def with_suffix(self, suffix: str) -> Self: ...
+    def relative_to(self, other: StrPath, *, walk_up: bool = False) -> Self: ...
+    def is_relative_to(self, other: StrPath) -> bool: ...
+    @property
+    def parts(self) -> tuple[str, ...]: ...
+    def joinpath(self, *pathsegments: StrPath) -> Self: ...
+    def __truediv__(self, key: StrPath) -> Self: ...
+    def __rtruediv__(self, key: StrPath) -> Self: ...
+    @property
+    def parent(self) -> Self: ...
+    @property
+    def parents(self) -> Sequence[Self]: ...
+    def is_absolute(self) -> bool: ...
+    def match(self, path_pattern: str, *, case_sensitive: bool | None = None) -> bool: ...
+    def full_match(self, pattern: StrPath, *, case_sensitive: bool | None = None) -> bool: ...
+
+class PathBase(PurePathBase):
+    def stat(self, *, follow_symlinks: bool = True) -> stat_result: ...
+    def lstat(self) -> stat_result: ...
+    def exists(self, *, follow_symlinks: bool = True) -> bool: ...
+    def is_dir(self, *, follow_symlinks: bool = True) -> bool: ...
+    def is_file(self, *, follow_symlinks: bool = True) -> bool: ...
+    def is_mount(self) -> bool: ...
+    def is_symlink(self) -> bool: ...
+    def is_junction(self) -> bool: ...
+    def is_block_device(self) -> bool: ...
+    def is_char_device(self) -> bool: ...
+    def is_fifo(self) -> bool: ...
+    def is_socket(self) -> bool: ...
+    def samefile(self, other_path: StrPath) -> bool: ...
+
+    # Adapted from builtins.open
```

Comment:
I don't know if we should require all implementations of this ABC to duplicate this stack of overloads. Not sure what the alternative would be, though.

Reply:
I hadn't considered that aspect when copying it over. I think the alternative is to define only the most generic form of open on the base class, and implementations can have more specific overloads as needed. I'll update the MR for that.

Reply:
stubcheck didn't like that... I thought it would work based on a simplified test I did locally, but possibly I misunderstood something. I'll need to look at it later, so I'll put this in draft for now.

Reply:
Also the same concern possibly applies to the overloads on ParserBase. I copied the definitions in the MR right now for that class from the versions that exist in posixpath.pyi.